tmpl/wiki/install.md

```diff
@@ -1,47 +1,86 @@
+* __10.9 or lower__: You will also need the [tuntap](http://tuntaposx.sourceforge.net/) kernel module if you want to use the MirageOS network stack from userspace.  Note that we do not test older versions of OSX beyond 10.10.
 
-    apt-get install build-essential m4 ocaml ocaml-native-compilers camlp4-extra
+If you are using Homebrew, run
 
-On CentOS 6.4, install the system compiler and libraries via:
+    brew install opam
+    opam init
+    opam install mirage
 
-    sudo yum groupinstall "Development Tools" "Development Libraries"
+### Ubuntu
 
-On Arch Linux, install the system compiler and libraries via:
-    sudo pacman -S ocaml camlp4 ocaml-findlib
-    yaourt -S opam
+#### Ubuntu 16.04 (Xenial) or higher
 
-Also note that the `mirage` configuration and deployment tool relies on the `xl` Xen toolstack to run virtual machines.
+This has the latest packages required in the base distribution, so just run:
 
-## Using OPAM
+    apt-get update
+    apt-get install opam
+    opam init
+    opam install mirage
 
-We use [OPAM](http://opam.ocaml.org) to manage OCaml compiler and library installations. It tracks library versions across upgrades and will recompile dependencies automatically if they get out of date. Please refer to OPAM [documentation](https://opam.ocaml.org) if you want to know more, but we will cover the basics to get you started here. Install OPAM for your operating system by following its [Quick Install Guide](http://opam.ocaml.org/doc/Install.html).
+#### Ubuntu 15.10 (Vivid) or lower
 
-Note that you require **OPAM 1.2 or greater** to use with MirageOS. Some distribution packages provide earlier versions and must be updated; check with
+The version of OPAM in older Ubuntus is not high enough to run Mirage (which requires OPAM 1.2.2 or higher), so you will need to add a custom PPA for the latest packages:
 
-    $ opam --version ## response should be at least 1.2.0, viz.
-    1.2.0
+    add-apt-repository ppa:avsm/ppa
+    apt-get update
+    apt-get install ocaml ocaml-native-compilers camlp4-extra opam
+    opam init
+    opam install mirage
 
-All the OPAM state is held in the `.opam` directory in your home directory, including compiler installations. You should never need to switch to a root user to install packages. Package listings are obtained through `remote` sources, which defaults to the contents of [github.com/ocaml/opam-repository](http://github.com/ocaml/opam-repository).
-After installation, `opam update -u` refreshes the package list and recompiles packages to the latest versions.
+Also note that the `mirage` configuration and deployment tool relies on the `xl` Xen toolstack to run Xen virtual machines.  Older Ubuntus may use the `xm` toolstack, so you will need to change it.
+
+### Debian 
+
+#### Debian Stable (Jessie)
+
+Debian Jessie only packages OPAM 1.2.0, but Mirage needs OPAM 1.2.2 or higher.  You can use [0install](http://0install.net) to get the right version of OPAM:
+
+    apt-get update
```

Comment:
Might be worth putting `sudo` before the `apt-get` commands (but not the others).